cirq/ops/linear_combinations_test.py

```diff
@@ -562,3 +562,101 @@ def test_linear_combination_of_operations_has_correct_pauli_expansion(
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    psum = pstr1 + pstr2
+    assert list(psum) == [pstr1, pstr2]
+
+    psum2 = cirq.PauliSum.from_pauli_strings([pstr1, pstr2])
+    assert psum == psum2
+
+
+def test_pauli_sum_from_single_pauli():
+    q = cirq.LineQubit.range(2)
+    psum1 = cirq.X(q[0]) + cirq.Y(q[1])
+    assert psum1 == cirq.PauliSum.from_pauli_strings([
+        cirq.X(q[0]) * 1, cirq.Y(q[1]) * 1
+    ])
+
+    psum2 = cirq.X(q[0]) * cirq.X(q[1]) + cirq.Y(q[1])
+    assert psum2 == cirq.PauliSum.from_pauli_strings([
+        cirq.X(q[0]) * cirq.X(q[1]), cirq.Y(q[1]) * 1
+    ])
+
+    psum3 = cirq.Y(q[1]) + cirq.X(q[0]) * cirq.X(q[1])
+    assert psum3 == psum2
+
+
+def test_pauli_sub():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    psum = pstr1 - pstr2
+
+    psum2 = cirq.PauliSum.from_pauli_strings([pstr1, -1 * pstr2])
+    assert psum == psum2
+
+
+def test_pauli_sub_simplify():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.X(q[0]) * cirq.X(q[1])
+    psum = pstr1 - pstr2
+
+    psum2 = cirq.PauliSum.from_pauli_strings([])
+    assert psum == psum2
+
+
+def test_paulisum_validation():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    with pytest.raises(ValueError) as e:
+        cirq.PauliSum([pstr1, pstr2])
+    assert e.match("Consider using")
+
+
+def test_add_number_paulisum():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    psum = cirq.PauliSum.from_pauli_strings([pstr1]) + 1.3
+    assert psum == cirq.PauliSum.from_pauli_strings([pstr1,
+                                                     cirq.PauliString({}, 1.3)])
+
+
+def test_add_number_paulistring():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    psum = pstr1 + 1.3
+    assert psum == cirq.PauliSum.from_pauli_strings([pstr1,
+                                                     cirq.PauliString({}, 1.3)])
+
+
+def test_pauli_sum_formatting():
+    q = cirq.LineQubit.range(2)
+    pauli = cirq.X(q[0])
+    assert str(pauli) == 'X(0)'
+    paulistr = cirq.X(q[0]) * cirq.X(q[1])
+    assert str(paulistr) == 'X(0)*X(1)'
+    paulisum1 = cirq.X(q[0]) * cirq.X(q[1]) + 4
+    assert str(paulisum1) == '1.000*X(0)*X(1)+4.000*I'
+    paulisum2 = cirq.X(q[0]) * cirq.X(q[1]) + cirq.Z(q[0])
+    assert str(paulisum2) == '1.000*X(0)*X(1)+1.000*Z(0)'
+    paulisum3 = cirq.X(q[0]) * cirq.X(q[1]) + cirq.Z(q[0]) * cirq.Z(q[1])
+    assert str(paulisum3) == '1.000*X(0)*X(1)+1.000*Z(0)*Z(1)'
+    assert "{:.0f}".format(paulisum3) == '1*X(0)*X(1)+1*Z(0)*Z(1)'
+
+    empty = cirq.PauliSum.from_pauli_strings([])
+    assert str(empty) == "0.000"
+
+
+def test_pauli_sum_repr():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    psum = pstr1 + pstr2
```

Comment:
Include a `*2` on the second term, and also include a `+1` identity term, to exercise more cases.

Reply:
done